unittests/resources/checks_unlisted/selfkill.py

```diff
@@ -25,7 +25,7 @@ def __init__(self):
         self.tags = {type(self).__name__}
         self.maintainers = ['TM']
 
-    def run(self):
-        super().run()
+    @rfm.run_before('run')
+    def self_kill(self):
```

Comment:
I think this test should be simply marked as "special." Using the hooks here changes its logic. The idea was to immediately kill ReFrame after a job is submitted. Here you rely too much on the sleeping time. In the past the sleepinh time was just to give it enough time to submit the job, but at least we knew that the kill would happen after the submit command.

Reply:
Then instead of using the `@rfm` decorator you could use:
`@pytest.mark.filterwarnings('ignore::reframe.core.exceptions.ReframeDeprecationWarning')`
on the `test_sigterm` test method

Reply:
@teojgo, no there is no need to do so. That's why we can mark a test as "special" with `special=True` in its definition.

Reply:
@vkarak yeap I have missed that

Reply:
I marked it as special